Source/Core/Core/HW/DVDInterface.cpp

```diff
@@ -1215,8 +1216,60 @@ DVDCommandResult ExecuteCommand(u32 command_0, u32 command_1, u32 command_2,
 // Simulates the timing aspects of reading data from a disc.
 // Returns the amount of ticks needed to finish executing the command,
 // and sets some state that is used the next time this function runs.
-u64 SimulateDiscReadTime(u64 offset, u32 length)
+//
+// This function does some extra handling for Wii discs on
+// top of what SimulateDiscReadTime does. When using a GC disc,
+// (or rather, when decrypt equals false and no data past
+// 4,7 GB is accessed,) both functions do the same thing.
+u64 SimulateDiscReadTimeWii(u64 offset, u32 length, bool decrypt)
 {
+	u64 end_offset = offset + length;
+
+	if (decrypt)
+	{
+		u64 data_start = VolumeHandler::GetDataStart();
+		offset = offset / 0x7C00 * 0x8000 + data_start;
+		if (length == 0)
+			end_offset = offset;
+		else
+			end_offset = (end_offset + 0x7BFF) / 0x7C00 * 0x8000 + data_start;
+	}
```

Comment:
I assume you're talking about the case where sequential reads access the same block? I assumed that it would be placed in the cache on the first read and then re-read from there, but now that I'm taking a closer look, the cache doesn't seem to work like that. Thanks for catching it.